File: Amazon/ClientSide/Amazon.ClientSide/src/app/Models/product.ts
import { Discount } from "./Discount";

export class Product {
    id: number
    name: string;
    description: string;
    price: number;
    pictureUrl: string;
    productImages: string[];
    categoryId: number;
    categoryName: string;
    brandId: number;
    brandName: string;
    quantityInStock: number;
    discount:Discount;
    sellerName?: string;

    constructor
        (
            id: number,
            name: string,
            price: number,
            description: string,
            pictureUrl: string,
            productImages: string[],
            categoryId: number,
            categoryName: string,
            brandId: number,
            brandName: string,
            quantityInStock: number,
            discount:Discount
        ) {
        this.id = id;
        this.name = name;
        this.price = price;
        this.description = description;
        this.pictureUrl = pictureUrl;
        this.productImages = productImages;
        this.categoryId = categoryId;
        this.categoryName = categoryName;
        this.brandId = brandId;
        this.brandName = brandName;
        this.quantityInStock = quantityInStock;
        this.discount = discount;
    }
}

File: Amazon/ServerSide/Amazon.Solution/Amazon.Core/Entities/Product.cs
using Microsoft.AspNetCore.Http;
using System.ComponentModel.DataAnnotations.Schema;
namespace Amazon.Core.Entities
{
    public class Product : BaseEntity
    {

        public string Name { get; set; }

        public string Description { get; set; }
        
        [Column(TypeName = "money")]
        public decimal Price { get; set; }

		public string PictureUrl { get; set; }

        [NotMapped]
		public IFormFile ImageFile { get; set; }

		public int QuantityInStock { get; set; }

        public virtual Category Category { get; set; }
        public int CategoryId { get; set; }

        public virtual Brand Brand { get; set; }
        public int BrandId { get; set; }

        public string SellerName { get; set; }
        public string SellerEmail { get; set; }

        public virtual ICollection<ProductImages> Images { get; set; } = [];
        [NotMapped]
        public ICollection<IFormFile> ImagesFiles { get; set; }

        public virtual Discount Discount { get; set; }

        public virtual ICollection<Review> Reviews { get; set; } = [];

	}
}
